lms/tasks/roster.py

```diff
@@ -53,11 +68,58 @@ def schedule_fetching_rosters() -> None:
                     no_recent_roster_clause,
                     recent_launches_cluase,
                 )
-                # Schedule only a few roster per call to this method
-                .limit(ROSTER_COURSE_LIMIT)
+                # Schedule only a few rosters per call to this method
+                .limit(ROSTER_LIMIT)
             )
             for lms_course_id in request.db.scalars(query).all():
-                fetch_roster.delay(lms_course_id=lms_course_id)
+                fetch_course_roster.delay(lms_course_id=lms_course_id)
+
+
+def schedule_fetching_assignment_rosters() -> None:
```

Comment:
Similar to the service code this is very similar to the course counterpart.

The query are different enough that trying to generalize resulting in code with lots of conditionals/parameters.